core-web/libs/portlets/edit-ema/portlet/src/lib/services/dot-page-api.service.ts

```diff
@@ -79,7 +85,18 @@ export class DotPageApiService {
         const url = params.url.replace(/^\/+|\/+$/g, '');
 
         const pageType = params.clientHost ? 'json' : 'render';
-        const apiUrl = `/api/v1/page/${pageType}/${url}?language_id=${params.language_id}&com.dotmarketing.persona.id=${params['com.dotmarketing.persona.id']}&mode=EDIT_MODE`;
+
+        const newQueryParams = buildQueryParams({
+            language_id: params.language_id ?? '1',
+            'com.dotmarketing.persona.id':
+                params['com.dotmarketing.persona.id'] ?? DEFAULT_PERSONA.identifier,
+            variantName: params.variantName ?? DEFAULT_VARIANT_ID,
+            mode: EDIT_MODE,
+            experimentId: params.experimentId
+        });
```

Comment:
Why is if this function called `buildQueryParams` why do we have so much logic in the params we're sending?

Reply:
I changed the name and some of the logic, maybe now it makes more sense.